core/src/cache.rs

```diff
@@ -1013,6 +1013,13 @@ impl Cache {
             .map(|TermEntry { state, .. }| std::mem::replace(state, new))
     }
 
+    /// Remove the cached term associated with this id.
+    ///
+    /// The file contents associated with this id remain, and they will be re-parsed if necessary.
+    pub fn reset(&mut self, file_id: FileId) {
```

Comment:
After discussion at the weekly, I'll refactor this so that `Cache`'s API does the rev-dependency invalidation. This way `Cache`'s public API can't leave things in an invalid state.